src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1456,6 +1462,45 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
     traverse(stats)
   }
 
+  private def typedUsecases(syms: List[Symbol], owner: Symbol)(implicit ctx: Context): Unit =
```

Comment:
I think expandParentDocs belongs in DottyDoc, not in Typer. So I would move typedUseCases there as well. Pass the typer as a paremeter, so it can call back to enterSymbol, createSymbol, typedStats.